runtime/interpreter/value.go

```diff
@@ -14996,13 +15027,25 @@ func newDictionaryValueFromOrderedMap(
 	staticType DictionaryStaticType,
 ) *DictionaryValue {
 
-	baseUse, lengthUse := common.NewDictionaryMemoryUsages(int(dict.Count()))
+	keySize := staticType.KeyType.elementSize()
+	valueSize := staticType.ValueType.elementSize()
```

Comment:
I see.  I think we check if `staticType != nil` in `NewArrayValue()`.  Do we want to remove it there?

Reply:
This is because some tests manually create array values without specifying a static type, so to preserve this behavior of the tests without crashing I added it there, but in actual programs it should never be `nil`.